activesupport/lib/active_support/test_case.rb

```diff
@@ -45,6 +46,15 @@ def test_order
 
         test_order
       end
+
+      def run(reporter, options = {})
+        if options[:patterns] && options[:patterns].any? { |p| p.include?(':') }
```

Comment:
Regarding the comment about [test-failures](https://github.com/rails/rails/pull/19571#issuecomment-94282124):

My guess is that `options[:patterns]` here is `["db:migrate", "test"]`. This does include a `:` resulting in a `CompositeFilter` being added.